### PR TITLE
Add apache license

### PR DESCRIPTION
**Pull request by charleskawczynski**
This PR adds the Apache 2 license, so that we register this package in order to smooth the ClimaCoupler workflow.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1 +1,202 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
Comment (cmbengue-ec-intl) [inline LICENSE:190]: update line 190: `Copyright  2022 California Institute of Technology`

**Review by cmbengue-ec-intl — COMMENTED** [review]
1. Include a [NOTICE file](https://github.com/CliMA/ClimaCore.jl/blob/main/NOTICE) to this PR.

File: LICENSE (hunk 1 above)
Comment (cmbengue) [inline LICENSE:190]: update line 190: `Copyright 2022 California Institute of Technology`

**Review by cmbengue — APPROVED** [review]
LGTM.

**Pull request merged**